spec/requests/users_controller_spec.rb

```diff
@@ -5225,6 +5225,84 @@ def users_found
     end
   end
 
+  describe "#lookup" do
+    fab!(:user_1) { Fabricate(:user) }
+    fab!(:user_2) { Fabricate(:user) }
+    fab!(:user_3) { Fabricate(:user) }
+
+    before { sign_in(user) }
+
+    it "lookups users by username" do
+      get "/u/lookup/users.json",
+          params: {
+            usernames: [user_1.username, user_2.username, user_3.username],
+          }
+
+      expect(response.status).to eq(200)
+      users = response.parsed_body["users"]
+      expect(users.length).to be(3)
+
+      ids = users.map { |user| user["id"] }
+      expect(ids).to include(user_1.id)
+      expect(ids).to include(user_2.id)
+      expect(ids).to include(user_3.id)
+    end
+
+    it "ignores non-existent usernames" do
+      get "/u/lookup/users.json",
+          params: {
+            usernames: [user_1.username, "unknown_username", "another_unknown_username"],
+          }
+
+      expect(response.status).to eq(200)
+      users = response.parsed_body["users"]
+      expect(users.length).to be(1)
+      expect(users[0]["id"]).to be(user_1.id)
+    end
+
+    it "returns users with ids, names and avatar_templates" do
+      get "/u/lookup/users.json", params: { usernames: [user_1.username] }
+
+      expect(response.status).to eq(200)
+      users = response.parsed_body["users"]
+
+      expect(users.length).to be(1)
+      expect(users[0]["id"]).to be(user_1.id)
+      expect(users[0]["username"]).to eq(user_1.username)
+      expect(users[0]["name"]).to eq(user_1.name)
+      expect(users[0]["avatar_template"]).to eq(user_1.avatar_template)
+    end
```

Comment:
I would also check we don't have OTHER attributes, like password for example, that would prevent any mistake in changing serializers.